src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/CSharp/ChatDetails.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Extensions.AI.Evaluation.Reporting;
+
+/// <summary>
+/// A class that records details related to all LLM chat conversation turns involved in the execution of a particular
+/// <see cref="ScenarioRun"/>.
+/// </summary>
+public sealed class ChatDetails
+{
+    /// <summary>
+    /// Gets or sets the <see cref="ChatTurnDetails"/> for the LLM chat conversation turns recorded in this
+    /// <see cref="ChatDetails"/> object.
+    /// </summary>
+#pragma warning disable CA2227
+    // CA2227: Collection properties should be read only.
+    // We disable this warning because we want this type to be fully mutable for serialization purposes and for general
+    // convenience.
+    public IList<ChatTurnDetails> TurnDetails { get; set; }
+#pragma warning restore CA2227
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ChatDetails"/> class.
+    /// </summary>
+    /// <param name="turnDetails">
+    /// A list of <see cref="ChatTurnDetails"/> objects.
+    /// </param>
+    [JsonConstructor]
+    public ChatDetails(IList<ChatTurnDetails> turnDetails)
+    {
+        TurnDetails = turnDetails;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ChatDetails"/> class.
+    /// </summary>
+    /// <param name="turnDetails">
+    /// An enumeration of <see cref="ChatTurnDetails"/> objects.
+    /// </param>
+    public ChatDetails(IEnumerable<ChatTurnDetails> turnDetails)
+        : this(turnDetails.ToList())
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ChatDetails"/> class.
+    /// </summary>
+    /// <param name="turnDetails">
+    /// An array of <see cref="ChatTurnDetails"/> objects.
+    /// </param>
+    public ChatDetails(params ChatTurnDetails[] turnDetails)
+        : this(turnDetails as IEnumerable<ChatTurnDetails>)
+    {
+    }
+
+    /// <summary>
+    /// Adds <see cref="ChatTurnDetails"/> for a particular LLM chat conversation turn to the <see cref="TurnDetails"/>
+    /// collection.
+    /// </summary>
+    /// <param name="turnDetails">
+    /// The <see cref="ChatTurnDetails"/> for a particular LLM chat conversation turn.
+    /// </param>
+    public void AddTurnDetails(ChatTurnDetails turnDetails)
+        => TurnDetails.Add(turnDetails);
```

Comment:
Is this necessary? It is extra public API surface, and it doesn't seem to offer additional value over `chatDetails.TurnDetails.Add()`.

Reply:
That is true and is a side effect of our earlier call to keep the result types in the reporting layer completely mutable (for convenience and to simplify the JSON serialization).

From the perspective of the type and IntelliSense, I think the method is (marginally) more discoverable. We also have similar APIs in other places (like `metric.AddDiagnostics()` instead of `metric.Diagnostics.Add()`). 

I'll leave this be for now but will try and take a holistic look at all similar occurrences in a follow-up PR. We can either remove the instance methods completely or convert these to convenience extension methods that live outside the type...